git-make-release

```diff
@@ -55,28 +73,52 @@ else
   echo "No previous tagged version found."
 fi
 
-# extract new version from CMakeLists.txt
+# extract version from CMakeLists.txt
 MAJOR=`grep -i 'set(${PROJECT_NAME}_MAJOR_VERSION' ${REPO_TOPLEVEL}/CMakeLists.txt | sed -e 's|^set(${PROJECT_NAME}_MAJOR_VERSION ||i' -e 's|)$||'`
 MINOR=`grep -i 'set(${PROJECT_NAME}_MINOR_VERSION' ${REPO_TOPLEVEL}/CMakeLists.txt | sed -e 's|^set(${PROJECT_NAME}_MINOR_VERSION ||i' -e 's|)$||'`
 PATCH=`grep -i 'set(${PROJECT_NAME}_PATCH_VERSION' ${REPO_TOPLEVEL}/CMakeLists.txt | sed -e 's|^set(${PROJECT_NAME}_PATCH_VERSION ||i' -e 's|)$||'`
 
+# check if we could extract the version from CMakeLists.txt, enter it manually otherwise
 if [ -z "$MAJOR" -o -z "$MINOR" -o -z "$PATCH" ]; then
   echo "Cannot detect version from CMakeLists.txt!"
+  if [ "${RELEASE_PATCH}" = "1" -o "${RELEASE_MINOR}" = "1" -o "${RELEASE_MAJOR}" = "1" ]; then
+    echo "Try again without the --patch, --minor or --major option to enter the version manually."
+    exit 1
+  fi
   echo -n "Enter new MAJOR version: "
   read MAJOR
   echo -n "Enter new MINOR version: "
   read MINOR
   echo -n "Enter new PATCH version: "
   read PATCH
-  if (( MAJOR < 10 )); then
-    MAJOR=0$MAJOR
-  fi
-  if (( MINOR < 10 )); then
-    MINOR=0$MINOR
-  fi
-  if (( PATCH < 10 )); then
-    PATCH=0$PATCH
-  fi
+fi
+
+# strip leading 0 from MAJOR, MINOR and PATCH
+MAJOR=${MAJOR#0}
+MINOR=${MINOR#0}
+PATCH=${PATCH#0}
+
+# calculate the next version if --patch, --minor or --major was given
+if [ "${RELEASE_PATCH}" = "1" ]; then
+  PATCH=$((PATCH + 1))
+elif [ "${RELEASE_MINOR}" = "1" ]; then
+  MINOR=$((MINOR + 1))
+  PATCH=0
+elif [ "${RELEASE_MAJOR}" = "1" ]; then
+  MAJOR=$((MAJOR + 1))
+  MINOR=0
+  PATCH=0
+fi
+
+# make sure the version numbers are two digits
+if (( MAJOR < 10 )); then
+  MAJOR=0$((MAJOR+0))
+fi
+if (( MINOR < 10 )); then
+  MINOR=0$((MINOR+0))
+fi
+if (( PATCH < 10 )); then
+  PATCH=0$((PATCH+0))
 fi
```

Comment:
I think `printf "%02d" $MAJOR` would be a bit shorter and without the calculation